### PR TITLE
[DCMAW-9786] Add dev platform assets to enable testing in codebuild

### PR DESCRIPTION
**Pull request by dothomson**
DCMAW-9786

This change includes building out the test container to be used in the dev, build and staging deployments. This is a pre-requisite for creating a post-merge Github action, which will be included in a subsequent PR.

### CODE REVIEW

File: backend-api/Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM node:iron-alpine AS builder
+
+WORKDIR /backend-api
+
+# Create a new user 'test' to avoid running the app as a root
+RUN adduser --disabled-password test
+RUN chown test .
+
+COPY package.json package-lock.json ./
+RUN npm install
```
Comment (sandy-may) [inline backend-api/Dockerfile.test:10]:
```suggestion
RUN npm install --ignore-scripts
```

**Review by sandy-may — COMMENTED** [review]
This looks great. Please may we make sure this is covered by Sonar? The sonar properties file at the root of the repo has a sonar.sources property that wouldn't capture this directory.

I think it should to make sure we've caught any docker/.sh codesmells

File: backend-api/Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM node:iron-alpine AS builder
```
Comment (sandy-may) [inline backend-api/Dockerfile.test:1]: question - is `iron` general knowledge as a major version of Node? 

I'm wondering whether it'd be clearer to people looking at these files to reference the version number instead as the base image?

Comment (sandy-may) [reply backend-api/Dockerfile.test:1]: Separately - do we need a builder here - it looks unused

Comment (dtrts) [reply backend-api/Dockerfile.test:1]: Ah yes, i started with builder but it didn't seem to be doing anything which wouldn't be brought
over to the final image. I have remove that. 


Re: `iron`, I matched the value from `.nvmrc`. I would change both if i was to change it here.

Comment (sandy-may) [reply backend-api/Dockerfile.test:1]: `iron` - understood :) Happy to keep with what you've done

File: backend-api/run-tests-locally.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -eu
+
+stack_name=${1:-mob-async-backend}
+
+echo Running test against "${stack_name}"
```
Comment (dothomson) [inline backend-api/run-tests-locally.sh:6]: Nit: 
```suggestion
echo "Running test against ${stack_name}"
```

Comment (sandy-may) [reply backend-api/run-tests-locally.sh:6]: is that a self-nit?

Comment (dothomson) [reply backend-api/run-tests-locally.sh]: Always working on becoming a better dev.

**Review by sandy-may — APPROVED** [review]

**Pull request merged**